src/main/java/graphql/execution/instrumentation/dataloader/PerLevelDataLoaderDispatchStrategy.java

```diff
@@ -508,11 +509,14 @@ void dispatch(int level, CallStack callStack) {
     }
 
 
-    public void dispatchDLCFImpl(Set<String> resultPathsToDispatch, Integer level, CallStack callStack) {
+    private void dispatchDLCFImpl(Set<String> resultPathsToDispatch, @Nullable Integer level, CallStack callStack) {
 
         // filter out all DataLoaderCFS that are matching the fields we want to dispatch
         List<ResultPathWithDataLoader> relevantResultPathWithDataLoader = new ArrayList<>();
-        for (ResultPathWithDataLoader resultPathWithDataLoader : callStack.allResultPathWithDataLoader) {
+        // we need to copy the list because the callStack.allResultPathWithDataLoader can be modified concurrently
+        // while iterating over it
+        ArrayList<ResultPathWithDataLoader> resultPathWithDataLoaders = new ArrayList<>(callStack.allResultPathWithDataLoader);
```

Comment:
@bbakerman fyi: this is a fix for a bug I found by running the tests on Java 21. This problem didn't show up earlier.